libraries/AC_Fence/AC_Fence.cpp

```diff
@@ -727,10 +709,17 @@ bool AC_Fence::check_fence_circle()
         return false;
     }
 
-    Vector2f home;
-    if (AP::ahrs().get_relative_position_NE_home(home)) {
+    Vector2f pos;
+    if (AP::ahrs().get_relative_position_NE_home(pos)) {
         // we (may) remain breached if we can't update home
-        _home_distance = home.length();
+        _home_distance = pos.length();
+
+        if (is_zero(pos.length_squared())) {
+            _circle_breach_direction.x = _circle_radius;
+            _circle_breach_direction.y = 0.0f;
+        } else {
+            _circle_breach_direction = pos.normalized() * _circle_radius.get()  - pos;
```

Comment:
if we recorded the position instead of an offset then we can calculate the offset in the get_breach_direction_NED() function then when the vehicle has moved we would be pointing to the right point